clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -688,9 +725,23 @@ public static ByteBuffer sizeDelimited(ByteBuffer buffer, int start) {
      * @param buffer Buffer containing the data to checksum
      * @param start Offset in the buffer to read from
      * @param size The number of bytes to include
+     * @return the computed checksum
      */
     public static long computeChecksum(ByteBuffer buffer, int start, int size) {
-        return Crc32.crc32(buffer.array(), buffer.arrayOffset() + start, size);
+        return computeChecksum(buffer.array(), buffer.arrayOffset() + start, size);
+    }
+
+    /**
+     * Compute the checksum of a range of data
+     * @param buffer Buffer containing the data to checksum
+     * @param start Offset in the buffer to read from
+     * @param size The number of bytes to include
+     * @return the computed checksum
+     */
+    public static long computeChecksum(byte[] buffer, int start, int size) {
+        Crc32 crc = new Crc32();
+        crc.update(buffer, start, size);
+        return crc.getValue();
```

Comment:
I think we can remove these utilities. I didn't know about the methods already in `Crc32` until I started updating that class.